app/renderer.js

```diff
@@ -107,10 +107,27 @@ module.exports = class Renderer {
     // ## Public methods
     // Renders given file with given options
     renderFile(files, options, done) {
+        let promise;
+
         if (!done && typeof options === "function") {
             done = options;
             options = {};
         }
+
+        if (!done && typeof Promise === 'function') {
+            promise = new Promise((resolve, reject) => {
+                done = (err) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve();
+                    }
+                };
+            });
+        }
+
+        done = done || function () {};
```

Comment:
It will introduce duplication I think.

```
if(done) done(err,res);
if(err || !res) reject(err);``` should be call on every error handler of the callback chain. But this way we just call `done(err)`.